projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.html

```diff
@@ -1,129 +1,153 @@
-    </mat-expansion-panel>
+                    <p fxLayout="row" fxLayoutAlign="space-between center" fxLayoutGap="0.5rem">
+                        <span>Colorizer:</span>
+                        <mat-button-toggle-group
+                            id="gradientType"
+                            fxFlex
+                            [multiple]="false"
+                            [ngModel]="getColorizerType()"
+                            (ngModelChange)="updateColorizerType($event)"
+                            [disabled]="getColorizerType() === 'palette'"
+                        >
+                            <mat-button-toggle fxFlex value="linearGradient">Linear Gradient</mat-button-toggle>
+                            <mat-button-toggle fxFlex value="logarithmicGradient">Logarithmic Gradient</mat-button-toggle>
+                            <mat-button-toggle fxFlex value="palette" [disabled]="true">Palette</mat-button-toggle>
+                        </mat-button-toggle-group>
+                    </p>
+                    <div
+                        *ngIf="symbology.colorizer.isGradient()"
+                        class="colorizer-preview"
+                        [style.background]="symbology.colorizer | geoengineColorizerCssGradient: 90"
+                    ></div>
+                </ng-template>
+            </mat-expansion-panel>
+
+            <mat-expansion-panel expanded="true">
+                <mat-expansion-panel-header>
+                    <mat-panel-title fxLayoutAlign="space-between center" fxLayoutGap="0.5rem">
+                        <mat-icon>looks</mat-icon>
+                        <span fxFlex>Update Color Map</span>
+                    </mat-panel-title>
+                </mat-expansion-panel-header>
 
-    <mat-expansion-panel expanded="true">
-        <mat-expansion-panel-header>
-            <mat-panel-title fxLayoutAlign="space-between center" fxLayoutGap="0.5rem">
-                <mat-icon>looks</mat-icon>
-                <span fxFlex>Update Color Map</span>
-            </mat-panel-title>
-        </mat-expansion-panel-header>
+                <ng-template matExpansionPanelContent>
+                    <div class="histogram">
+                        <ng-container *ngIf="histogramData | async as histogramData">
+                            <geoengine-vega-viewer
+                                [chartData]="histogramData"
+                                (interactionChange)="updateBounds($any($event))"
+                            ></geoengine-vega-viewer>
+                        </ng-container>
+                        <mat-progress-bar mode="indeterminate" *ngIf="histogramLoading | async"></mat-progress-bar>
+                    </div>
+                    <div>
+                        <mat-slide-toggle [checked]="histogramAutoReload" (change)="histogramAutoReload = $event.checked">
+                            Sync map and histogram
+                        </mat-slide-toggle>
```

Comment:
You still need to change this to only compute the histogram when you click a "Calculate histogram" button.

Reply:
Changed it to a button now f2c0f44af28f79d4d5bb91812260706d2c463892

Reply:
The important thing is, that it is *not* calculated automatically upfront.
The user has to calculate it actively so that the user can decide whether to do it or not.
This is why I meant "Calculate histogram" since you cannot update it upfront.

Reply:
I removed the automatic calculation and changed the button text to "Calculate Histogram" df5f7fdfc5f3388cdcb12b1a05092ac86301c885